### PR TITLE
MGMT-2858: LogSender - use coredumpctl

### PR DESCRIPTION
**Pull request by danielerez**
Used coredumpctl to fetch all coresumps.

### CODE REVIEW

File: src/logs_sender/send_logs.go [hunk 1]
```diff
@@ -136,21 +136,20 @@ func getDmesgLogs(l LogsSender, outputFilePath string) error {
 
 func getCoreDumps(l LogsSender, targetDir string) error {
 	log.Infof("Get coredump files")
-	coredumpPath := "/var/lib/systemd/coredump/"
-	stdout, stderr, exitCode := l.Execute("ls", coredumpPath)
+	stdout, _, exitCode := l.Execute("coredumpctl", "list", "--no-legend")
 	if exitCode != 0 {
```
Comment (osherdp) [inline src/logs_sender/send_logs.go:140]: It doesn't return specific ``exitCode`` when there are not matches?
We won't return "No coredump files available" on legit errors?

File: src/logs_sender/send_logs.go [hunk 2]
```diff
@@ -136,21 +136,20 @@ func getDmesgLogs(l LogsSender, outputFilePath string) error {
...
 	}
 
-	files := strings.Split(strings.TrimSuffix(stdout, "\n"), "\n")
-	for _, file := range files {
-		outputFile := path.Join(targetDir, fmt.Sprintf("coredump_%s", file))
-		stderr, exitCode := l.ExecuteOutputToFile(outputFile, "cat", coredumpPath + file)
+	dumps := strings.Split(strings.TrimSuffix(stdout, "\n"), "\n")
+	for _, dump := range dumps {
+		pid := strings.Fields(dump)[4]
+		outputFile := path.Join(targetDir, fmt.Sprintf("coredump_pid_%s", pid))
```
Comment (osherdp) [inline src/logs_sender/send_logs.go:148]: is it possible to embed the executable basename path as well in the target path?

**Review by osherdp — CHANGES_REQUESTED** [review]

File: src/logs_sender/send_logs.go (hunk 1 above)
Comment (danielerez) [reply src/logs_sender/send_logs.go:140]: It returns a generic error(1) so I'll just add stderr to the log.

File: src/logs_sender/send_logs.go (hunk 2 above)
Comment (danielerez) [reply src/logs_sender/send_logs.go:148]: sure, added to the filename

**Pull request merged**